### DISCUSSION TITLE
Конфигурация для csharp_prefer_braces

### DISCUSSION THREAD

**Discussion by FrediKats** [kysect/CodeRules#19] [General]
Сейчас в editorconfig стоит `csharp_prefer_braces = when_multiline:warning`. Было предложение заменить на always. Это значит, что мы будем форсить 

```csharp
if (value)
{
  Do();
}

// instead of

if (value)
  Do();
```

Нужно опредилиться с ожидаемым поведением.

**Comment by I-SER-I** [comment]
Я где-то видел еще такое
```cs
if (value) throw new Exception();
```

**Reply by FrediKats** [reply]
Я очень не люблю такие строчки т.к. в одной строке и условие и логика. Но понимаю, что для валидации в начале метода оно выглядит лучше. И получается ступор.

**Comment by ronimizy** [comment]
Моё мнение на этот счёт, что always стиль очень сильно раздувает количество кода, без объективного преимущества в читаемости и красоте. Если выражение **по настоящему** singleline, то оно будет нормально читаться без скобок, если нет, то даже со скобками будет отвратительно, в таком случае нужно вынести значения в переменные, или, если это не помогает, например они и так в переменных, так же ставить скобки.
